src/System.Private.CoreLib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -51,6 +51,9 @@ private struct Entry
 
         private int[]? _buckets;
         private Entry[]? _entries;
+#if BIT64
+        private ulong _fastModMultiplier;
```

Comment:
I wish we had a good way to reason about these type of changes that add a bit of cost on one end and save a bit on the other end.

Reply:
The ratio of `.ContainsKey`,`.TryGetValue` and even `.Add` should be far higher than events that cause resize?

I also hope that FastMod can have the support check dropped and be changed to 
```csharp
public static uint FastMod(uint value, uint divisor, ulong multiplier)
{
    ulong lowbits = multiplier * value;
    return (uint)Math.MultiplyHigh(lowbits, divisor);
}
```

Reply:
I guess the concern here is increased memory consumption.

Reply:
Right